src/bundle/Resources/views/themes/admin/content/page_title_edit.html.twig

```diff
@@ -2,9 +2,10 @@
 
 {% block close_button %}
     {% if without_close_button is not defined or without_close_button != true %}
+        {% set referrer_location = parent_location.id > 1 ? parent_location : location %}
```

Comment:
This will redirect to the location only when the user edits something under root location, what we need is restore the behavior from 3.1:
When the user creates content under some location he is redirected to the parent location when he closes the create mode.
When the user edits content he is redirected to this location, not to the parent.